src/System/Random/Monad.hs

```diff
@@ -420,7 +423,37 @@ runSTGen_ g action = fst $ runSTGen g action
 -- *   On the other hand, there is no correct way to construct a
 --     @UniformRange (a, b)@ instance based on just @UniformRange a@ and
 --     @UniformRange b@.
-
+--
+-- $floating
+--
+-- The 'UniformRange' instances for 'Float' and 'Double' use the following
+-- procedure to generate random value in a range for @uniformRM (a, b) g@:
+--
+-- 1.  Generate \(x\) uniformly such that \(0 \leq x \leq 1\).
+--
+-- 2.  Return \((b - a) * x + a\).
+--
+-- As a result, the following caveats apply:
+--
+-- *   The method by which \(x\) is sampled does not cover all representable
+--     floating point numbers in that interval. The method never generates
+--     denormal floating point numbers, for example.
+--
+-- *   Due to rounding errors, @uniformRM (a, b)@ can generate values outside
+--     the \([a, b]\) range for pathological inputs, for example:
+--
+--     >>> :{
+--     let a = -4.7021254e-38
+--         b = -1.481e-42
+--         x = 1.0
+--         result = (b - a) * x + a :: Float
+--     in (result, result <= b)
+--     :}
+--     (-1.48e-42,False)
```

Comment:
Q: is this doctest example useful or a distraction?

Reply:
I think it's a distraction

> I am tempted say that for any a and b, be they NaN, Infinity, normal or denormal, you get precisely the semantics of (b - a) * x + b. That is concise and covers all the edge cases. Perhaps we could include a few examples.

I like this and one example could be the above where the result is outside the bounds.

Reply:
Done.